pages/treeherder.py

```diff
@@ -52,6 +52,81 @@ def open_perfherder_page(self):
         return PerfherderPage(self.base_url, self.selenium).wait_for_page_to_load()
 
 
+    class Result(PageRegion):
+
+        _job_result_status_locator = (By.CSS_SELECTOR, '#result-status-pane > div:nth-child(1) > span')
+        _result_set_locator = (By.CSS_SELECTOR, '#th-global-content .result-set')
+        _selected_job_title_locator = (By.CSS_SELECTOR, '.job-list .selected-job')
+
+        @property
+        def job_result_status(self):
+            return self.selenium.find_element(*self._job_result_status_locator).text
+
+        @property
+        def result(self):
+            return [Result(self, root=el) for el in
+                self.find_elements(self._result_set_locator)]
+
+        def pin_using_spacebar(self):
+            el = self.selenium.find_element(*self._result_set_locator)
+            Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+            el.send_keys(Keys.SPACE)
+            Wait(self.selenium, self.timeout).until(lambda s: self.pinboard.is_pinboard_open)
+
+        def select_next_job(self):
+            el = self.selenium.find_element(*self._result_set_locator)
+            Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+            el.send_keys(Keys.ARROW_RIGHT)
+            Wait(self.selenium, self.timeout).until(lambda s: self.job_result_status)
+            return self.selenium.find_element(*self._selected_job_title_locator).get_attribute('title')
+
+    class JobDetails(PageRegion):
+
+        _job_details_actionbar_locator = (By.ID, 'job-details-actionbar')
+        _job_details_pin_job_locator = (By.CSS_SELECTOR, '#job-details-actionbar .nav .nav li:nth-child(1) a')
+        _logviewer_button_locator = (By.ID, 'logviewer-btn')
+        _result_status_locator = (By.ID, 'job-details-panel')
+
+        def open_logviewer(self):
+            Wait(self.selenium, self.timeout).until(
+                EC.visibility_of_element_located(self._job_details_actionbar_locator))
+            self.selenium.find_element(*self._result_set_locator).send_keys('l')
+            return LogviewerPage(self.base_url, self.selenium)
+
+        def pin_job(self):
+            el = Wait(self.selenium, self.timeout).until(
+                EC.visibility_of_element_located(self._job_details_actionbar_locator))
+            el.find_element(*self._job_details_pin_job_locator).click()
+            from pages.treeherder.pinboard import Pinboard
+            return Pinboard(self.base_url, self.selenium)
+
+
+    class Pinboard(PageRegion):
```

Comment:
It's a good practice for page regions to define a root locator. In this case I believe it would be:

```python
_root_locator = (By.ID, 'pinboard-panel')
```